### DISCUSSION TITLE
Preparing for VulkanSceneGraph-1.1.8 - test result for Jetson Nano with Ubuntu 18.04

### DISCUSSION THREAD

**Discussion by reedev** [vsg-dev/VulkanSceneGraph#1283] [General]
Hi Robert,

I ran the same list of examples on a Jetson. As you know, unfortunately, NVIDIA does not upgrade the OS on the Jetson...

Vulkan Instance Version: 1.2.70

NVIDIA Tegra X1 (nvgpu)

Bus error for the following examples, the rest seems to run ok:

vsgmultiviews
vsgviewer models/openstreetmap.vsgt
vsghelloworld
vsgrendertotexture models/openstreetmap.vsgt
vsgwindows models/openstreetmap.vsgt
vsgoffscreenshot models/openstreetmap.vsgt
vsgheadless models/openstreetmap.vsgt
vsgskybox models/openstreetmap.vsgt
vsgscreenshot models/openstreetmap.vsgt
vsgoverlay models/openstreetmap.vsgt
vsgallocator models/openstreetmap.vsgt
vsglog_mt
vsgtransform models/openstreetmap.vsgt
vsgpagedlod
vsgtextureprojection textures/VSGlogo.png
vsglayers models/openstreetmap.vsgt
vsgtiledatabase
vsgannotation models/openstreetmap.vsgt
vsgdynamicvertex models/openstreetmap.vsgt
vsgclip models/openstreetmap.vsgt
vsgcomputevertex
vsgdynamicviews
vsgdynamicload models/openstreetmap.vsgt
vsgdynamicwindows
vsginstrumentation models/openstreetmap.vsgt

Before I was using v1.1.3 but had not run all the examples, I will do that asap.

**Comment by robertosfield** [comment]
Do all these examples that fail, fail in the same way?  You mention a bus error.  Could you provide a stack trace for a couple of these failing examples as this might help shine a light to what might be going on.

**Comment by reedev** [comment]
For now, I checked 2 examples which don't rely on any data, and to me they don't seem (exactly) the same to me:

Reading symbols from vsghelloworld...done.
(gdb) run
Starting program: /home/fdpn/src/extern_build/vsgExamples-master/bin/vsghelloworld 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/aarch64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fad12a6e0 (LWP 6481)]
[New Thread 0x7fac6a96e0 (LWP 6482)]
[New Thread 0x7fabea86e0 (LWP 6483)]

Thread 3 "vsghelloworld" received signal SIGBUS, Bus error.
[Switching to Thread 0x7fac6a96e0 (LWP 6482)]
__pthread_cond_wait_common (abstime=0x7fac6a8cb8, mutex=0x7fb0a8cda4, cond=0x7fb0a8cdd4) at pthread_cond_wait.c:397
397     pthread_cond_wait.c: No such file or directory.
(gdb) 
(gdb) 
(gdb) bt
#0  __pthread_cond_wait_common (abstime=0x7fac6a8cb8, mutex=0x7fb0a8cda4, cond=0x7fb0a8cdd4) at pthread_cond_wait.c:397
#1  __pthread_cond_timedwait (cond=0x7fb0a8cdd4, mutex=0x7fb0a8cda4, abstime=0x7fac6a8cb8) at pthread_cond_wait.c:667
#2  0x000000555572c4b0 in __gthread_cond_timedwait (__abs_timeout=0x7fac6a8cb8, __mutex=0x7fb0a8cda4, __cond=0x7fb0a8cdd4)
    at /usr/include/aarch64-linux-gnu/c++/8/bits/gthr-default.h:871
#3  std::condition_variable::__wait_until_impl<std::chrono::duration<long, std::ratio<1l, 1000000000l> > > (__atime=..., __lock=<synthetic pointer>..., 
    this=0x7fb0a8cdd4) at /usr/include/c++/8/condition_variable:178
#4  std::condition_variable::wait_until<std::chrono::duration<long, std::ratio<1l, 1000000000l> > > (__atime=..., __lock=<synthetic pointer>..., this=0x7fb0a8cdd4)
    at /usr/include/c++/8/condition_variable:106
#5  std::condition_variable::wait_for<long, std::ratio<1l, 1000l> > (__rtime=<synthetic pointer>..., __lock=<synthetic pointer>..., this=0x7fb0a8cdd4)
    at /usr/include/c++/8/condition_variable:143
#6  vsg::DatabaseQueue::take_when_available (this=0x7fb0a8cd8c) at /home/fdpn/VSG/VulkanSceneGraph-master/src/vsg/io/DatabasePager.cpp:63
#7  0x000000555572f780 in vsg::DatabasePager::<lambda(vsg::ref_ptr<vsg::DatabaseQueue>, vsg::ref_ptr<vsg::ActivityStatus>, vsg::DatabasePager&, const string&)>::operator()(vsg::ref_ptr<vsg::DatabaseQueue>, vsg::ref_ptr<vsg::ActivityStatus>, vsg::DatabasePager &, const std::(char, long long, long long, ::string &)) (
    requestQueue=..., status=..., databasePager=..., threadName=..., __closure=<optimized out>)
    at /home/fdpn/VSG/VulkanSceneGraph-master/src/vsg/io/DatabasePager.cpp:150
#8  0x000000555573085c in std::__invoke_impl<void, vsg::DatabasePager::start(uint32_t)::<lambda(vsg::ref_ptr<vsg::DatabaseQueue>, vsg::ref_ptr<vsg::ActivityStatus>, vsg::DatabasePager&, const string&)>, std::reference_wrapper<vsg::ref_ptr<vsg::DatabaseQueue> >, std::reference_wrapper<vsg::ref_ptr<vsg::ActivityStatus> >, std::reference_wrapper<vsg::DatabasePager>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > (__f=...)
    at /usr/include/c++/8/bits/refwrap.h:308
#9  std::__invoke<vsg::DatabasePager::start(uint32_t)::<lambda(vsg::ref_ptr<vsg::DatabaseQueue>, vsg::ref_ptr<vsg::ActivityStatus>, vsg::DatabasePager&, const string&)>, std::reference_wrapper<vsg::ref_ptr<vsg::DatabaseQueue> >, std::reference_wrapper<vsg::ref_ptr<vsg::ActivityStatus> >, std::reference_wrapper<vsg::DatabasePager>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > (__fn=...) at /usr/include/c++/8/bits/invoke.h:95
#10 std::thread::_Invoker<std::tuple<vsg::DatabasePager::start(uint32_t)::<lambda(vsg::ref_ptr<vsg::DatabaseQueue>, vsg::ref_ptr<vsg::ActivityStatus>, vsg::DatabasePager&, const string&)>, std::reference_wrapper<vsg::ref_ptr<vsg::DatabaseQueue> >, std::reference_wrapper<vsg::ref_ptr<vsg::ActivityStatus> >, std::reference_wrapper<vsg::DatabasePager>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::_M_invoke<0, 1, 2, 3, 4> (this=<optimized out>)
    at /usr/include/c++/8/thread:244
#11 std::thread::_Invoker<std::tuple<vsg::DatabasePager::start(uint32_t)::<lambda(vsg::ref_ptr<vsg::DatabaseQueue>, vsg::ref_ptr<vsg::ActivityStatus>, vsg::DatabasePager&, const string&)>, std::reference_wrapper<vsg::ref_ptr<vsg::DatabaseQueue> >, std::reference_wrapper<vsg::ref_ptr<vsg::ActivityStatus> >, std::reference_wrapper<vsg::DatabasePager>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::operator() (this=<optimized out>)
    at /usr/include/c++/8/thread:253
#12 std::thread::_State_impl<std::thread::_Invoker<std::tuple<vsg::DatabasePager::start(uint32_t)::<lambda(vsg::ref_ptr<vsg::DatabaseQueue>, vsg::ref_ptr<vsg::ActivityStatus>, vsg::DatabasePager&, const string&)>, std::reference_wrapper<vsg::ref_ptr<vsg::DatabaseQueue> >, std::reference_wrapper<vsg::ref_ptr<vsg::ActivityStatus> >, std::reference_wrapper<vsg::DatabasePager>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >::_M_run(void) (
    this=<optimized out>) at /usr/include/c++/8/thread:196
#13 0x0000007fb64d3e94 in ?? () from /usr/lib/aarch64-linux-gnu/libstdc++.so.6
#14 0x0000007fb7f10088 in start_thread (arg=0x7fffffdacf) at pthread_create.c:463
#15 0x0000007fb62b20cc in thread_start () at ../sysdeps/unix/sysv/linux/aarch64/clone.S:78
(gdb) 





Reading symbols from vsgtiledatabase...done.
(gdb) run
Starting program: /home/fdpn/src/extern_build/vsgExamples-master/bin/vsgtiledatabase 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/aarch64-linux-gnu/libthread_db.so.1".

[New Thread 0x7fad12a6e0 (LWP 6927)]
[New Thread 0x7fac6a96e0 (LWP 6928)]
[New Thread 0x7fabea86e0 (LWP 6929)]
[New Thread 0x7fab6a76e0 (LWP 6930)]
[New Thread 0x7faaea66e0 (LWP 6931)]
[New Thread 0x7faa6a56e0 (LWP 6932)]

Thread 1 "vsgtiledatabase" received signal SIGBUS, Bus error.
0x0000005555774510 in std::__atomic_base<unsigned long>::exchange (__m=std::memory_order_seq_cst, __i=0, this=0x7fb0a8d274)
    at /usr/include/c++/8/bits/atomic_base.h:410
410           exchange(__int_type __i,
(gdb) 
(gdb) bt
#0  0x0000005555774510 in std::__atomic_base<unsigned long>::exchange (__m=std::memory_order_seq_cst, __i=0, this=0x7fb0a8d274)
    at /usr/include/c++/8/bits/atomic_base.h:410
#1  vsg::DatabasePager::updateSceneGraph (this=0x7fb0a8d24c, frameStamp=..., cr=...) at /home/fdpn/VSG/VulkanSceneGraph-master/src/vsg/io/DatabasePager.cpp:264
#2  0x000000555569357c in vsg::Viewer::update (this=0x7fb0a87140) at /usr/include/c++/8/bits/atomic_base.h:512
#3  0x00000055556591e8 in main (argc=<optimized out>, argv=<optimized out>)
    at /home/fdpn/VSG/vsgExamples-master/examples/nodes/vsgtiledatabase/vsgtiledatabase.cpp:253
(gdb) 

Of course, I can run each example via gdb. Please let me know how we can proceed.
